diff -ruN collectd-4.10.3.orig/src/Makefile.am collectd-4.10.3/src/Makefile.am
--- collectd-4.10.3.orig/src/Makefile.am	2011-03-26 17:08:53.000000000 +0100
+++ collectd-4.10.3/src/Makefile.am	2011-04-24 22:56:39.521229944 +0200
@@ -7,7 +7,7 @@
 PLUGIN_LDFLAGS = -module -avoid-version -export-symbols-regex '\<module_register\>'
 
 if COMPILER_IS_GCC
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 endif
 
 AM_CPPFLAGS = -I$(srcdir)/daemon
diff -ruN collectd-4.10.3.orig/src/libcollectdclient/Makefile.am collectd-4.10.3/src/libcollectdclient/Makefile.am
--- collectd-4.10.3.orig/src/libcollectdclient/Makefile.am	2011-03-26 17:04:26.000000000 +0100
+++ collectd-4.10.3/src/libcollectdclient/Makefile.am	2011-04-24 22:57:12.231229943 +0200
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 
 if COMPILER_IS_GCC
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 endif
 
 pkginclude_HEADERS = collectd/client.h collectd/network.h collectd/network_buffer.h collectd/lcc_features.h
